Mathlib/Topology/Algebra/ClopenNhdofOne.lean

```diff
@@ -0,0 +1,29 @@
+/-
+Copyright (c) 2024 Nailin Guan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nailin Guan, Yi Song, Xuchun Li
+-/
+import Mathlib.GroupTheory.Index
+import Mathlib.Topology.Algebra.ClosedSubgroup
+import Mathlib.Topology.Algebra.OpenSubgroup
+/-!
+# Open normal subgroup in a clopen neighborhood of One
+This file defines `OpenNormalSubgroupSubClopenNhdsOfOne`, which strengthens the result of
+`OpenSubgroupSubClopenNhdsOfOne` into open *normal* subgroups.
+
+This file is split out from the file `OpenSubgroup` because the need of more imports.
+-/
+
+namespace TopologicalGroup
+
+theorem existOpenNormalSubgroupSubClopenNhdsOfOne {G : Type*} [Group G] [TopologicalSpace G]
+    [TopologicalGroup G] [CompactSpace G] {W : Set G} (WClopen : IsClopen W) (einW : 1 ∈ W) :
+    ∃ H : OpenNormalSubgroup G, (H : Set G) ⊆ W := by
+  rcases existOpenSubgroupSubClopenNhdsOfOne WClopen einW with ⟨H, hH⟩
+  letI : Subgroup.FiniteIndex H.1 := Subgroup.finiteIndex_of_finite_quotient H.1
```

Comment:
Why `letI` instead of `have`? This is a `Prop`, and inlining brings nothing here.

Reply:
Sorry, what do you mean "inlining brings nothing here"?

Reply:
Do you know the difference between `have` and `haveI`? It’s all about inlining (the letter `I` stands for inlining). So if you don’t know what inlining means then you should never use `haveI` or `letI`. See the explanations in [core Lean](https://github.com/leanprover/lean4/blob/ac80e261bd253e0d41de5027d5e8d25570b7f982/src/Init/Tactics.lean#L1125-L1129).

Note that, very confusingly, Lean 3 used to have tactics called `haveI` and `letI` that were completely unrelated to this (at that time the letter `I` stood for “instance”). So if you learned this in Lean 3, you simply need to forget about it. Those Lean 3 tactics are no longer needed in Lean 4.

Reply:
Thank you, this should be changed into `have`